### PR TITLE
AdMob AdRequest Neighboring Content URLS

### PR DESCRIPTION
**Pull request by DellaBitta**
Adds the ability to configure Neighboring Content URL strings to the AdRequest object.
Additionally renamed the parameter in the `AdRequest::add_extra` method from `add_extra(const char* ad_network, …)` to `add_extra(const char* adapter_class_name, … )` to match nomenclature in the iOS and Android SDKs.

***

[Integration Tests CI](https://github.com/firebase/firebase-cpp-sdk/actions/runs/1526084026)
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [X] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: app/src/util_ios.h [hunk 1]
```diff
@@ -193,10 +194,14 @@ void ForEachAppDelegateClass(void (^block)(Class));
 NSMutableArray *StringVectorToNSMutableArray(
     const std::vector<std::string> &vector);
 
+// Convert a unordered_set of strings into an NSMutableArray.
+NSMutableArray *StringUnorderedSetToNSMutableArray(
+    const std::unordered_set<std::string> &set);
```
[the remark this answers is not in the document]
Comment (DellaBitta) [reply app/src/util_ios.h:199]: I did!

File: app/src/util_ios.h [hunk 2]
```diff
@@ -193,10 +194,14 @@ void ForEachAppDelegateClass(void (^block)(Class));
...
+
 // Convert a NSArray into a vector of strings.  Asserts if a non NSString
 // object is found in the array.
-void NSArrayOfNSStringToVectorOfString(NSArray* array,
-    std::vector<std::string>* string_vector);
+void NSArrayOfNSStringToVectorOfString(
+  NSArray *array, std::vector<std::string> *string_vector);
```
[the remark this answers is not in the document]
Comment (DellaBitta) [reply app/src/util_ios.h:204]: `#include <vector>` is indeed at the top of the file.

File: admob/integration_test/src/integration_test.cc [hunk 3]
```diff
@@ -120,6 +120,11 @@ static const char* kAdNetworkExtrasInvalidClassName = "abc123321cba";
 
 static const char* kContentUrl = "http://www.firebase.com";
 
+static const std::vector<std::string> kNeighboringContentURLs = {
+    "https://firebase.google.com/products-build",
```
Comment (jonsimantov) [inline admob/integration_test/src/integration_test.cc:124]: Are these real URLs? should we use fake ones?

Comment (DellaBitta) [reply admob/integration_test/src/integration_test.cc]: Done.

**Review by jonsimantov — APPROVED** [review]
SGTM, as long as the underlying SDK doesn't mess up if it expects URLs and gets these instead. :)

File: app/src/util_ios.h [hunk 4]
```diff
@@ -24,6 +24,7 @@
 
 #include <map>
 #include <string>
+#include <unordered_set>
 #include <vector>
 
 #include "app/src/include/firebase/variant.h"
```
Comment (jonsimantov) [inline app/src/util_ios.h:30]: I think you can fix the lint error if you move the system #includes above the #ifdef __OBJC__.

**Pull request merged**